src/Microsoft.OData.Core/UriParser/TreeNodeKinds/QueryNodeKind.cs

```diff
@@ -158,6 +158,11 @@ public enum QueryNodeKind
         /// Count of a collection contains primitive or enum or complex or entity type.
         /// </summary>
         Count = InternalQueryNodeKind.Count,
+
+        /// <summary>
+        /// Cast on a single property.
+        /// </summary>
+        SinglePropertyCast = InternalQueryNodeKind.SinglePropertyCast,

```

Comment:
the cast works on complex properties as well as primitive properties.

---
In reply to: [125081678](https://github.com/OData/odata.net/pull/848#discussion_r125081678) [](ancestors = 125081678,124892629)